ReactNativeClient/lib/services/searchengine/SearchEngine.js

```diff
@@ -432,54 +448,45 @@ class SearchEngine {
 		return SearchEngine.SEARCH_TYPE_FTS;
 	}
 
-	async search(query, options = null) {
+	async search(searchString, options = null) {
 		options = Object.assign({}, {
 			searchType: SearchEngine.SEARCH_TYPE_AUTO,
 		}, options);
 
-		query = this.normalizeText_(query);
+		searchString = this.normalizeText_(searchString);
 
-		const searchType = this.determineSearchType_(query, options.searchType);
-		const parsedQuery = this.parseQuery(query);
+		const searchType = this.determineSearchType_(searchString, options.searchType);
 
 		if (searchType === SearchEngine.SEARCH_TYPE_BASIC) {
 			// Non-alphabetical languages aren't support by SQLite FTS (except with extensions which are not available in all platforms)
-			const rows = await this.basicSearch(query);
+			const rows = await this.basicSearch(searchString);
+			const parsedQuery = this.parseQuery(searchString);
 			this.processResults_(rows, parsedQuery, true);
 			return rows;
-		} else { // SEARCH_TYPE_FTS
+		} else {
+			// SEARCH_TYPE_FTS
 			// FTS will ignore all special characters, like "-" in the index. So if
 			// we search for "this-phrase" it won't find it because it will only
 			// see "this phrase" in the index. Because of this, we remove the dashes
 			// when searching.
 			// https://github.com/laurent22/joplin/issues/1075#issuecomment-459258856
-			query = query.replace(/-/g, ' ');
-
-			// Note that when the search engine index is somehow corrupted, it might contain
-			// references to notes that don't exist. Not clear how it can happen, but anyway
-			// handle it here by checking if `user_updated_time` IS NOT NULL. Was causing this
-			// issue: https://discourse.joplinapp.org/t/how-to-recover-corrupted-database/9367
-			const sql = `
-				SELECT
-					notes_fts.id,
-					notes_fts.title AS normalized_title,
-					offsets(notes_fts) AS offsets,
-					notes.title,
-					notes.user_updated_time,
-					notes.is_todo,
-					notes.todo_completed,
-					notes.parent_id
-				FROM notes_fts
-				LEFT JOIN notes ON notes_fts.id = notes.id
-				WHERE notes_fts MATCH ?
-				AND notes.user_updated_time IS NOT NULL
-			`;
+
+			// commented out because of search syntax like day-0 or -tag:something we can't do this.
+			// searchString = searchString.replace(/-/g, ' ')
```

Comment:
This was meant to address this issue: https://github.com/laurent22/joplin/issues/1075#issuecomment-459258856 How does your current code handle this? And if it does already handle it well I think you can simply delete this code and associated comment.

Reply:
If the word is quoted, then FTS replaces the line separators by itself. So `"does-not_work"` will actually search for `"does not work"`. So I'm quoting all words if not already quoted. 

This won't matter to the search semantics. 
Eg. Searching for `"does"` and `does` are the same.